packages/backend-core/src/environment.ts

```diff
@@ -208,6 +209,32 @@ const environment = {
   OPENAI_API_KEY: process.env.OPENAI_API_KEY,
 }
 
+export function setEnv(newEnvVars: Partial<typeof environment>): () => void {
+  const oldEnv = cloneDeep(environment)
+
+  let key: keyof typeof newEnvVars
+  for (key in newEnvVars) {
+    environment._set(key, newEnvVars[key])
+  }
+
+  return () => {
+    for (const [key, value] of Object.entries(oldEnv)) {
+      environment._set(key, value)
+    }
+  }
+}
+
+export function withEnv<T>(envVars: Partial<typeof environment>, f: () => T) {
+  const cleanup = setEnv(envVars)
+  const result = f()
+  if (result instanceof Promise) {
+    return result.finally(cleanup)
+  } else {
+    cleanup()
+    return result
+  }
+}
```

Comment:
This is only used in tests, and exists to the mirror the functionality we have in `TestConfiguration` in `packages/server`. I'm not a big fan of it lying around like this but wasn't sure where else to put it.

Reply:
Can we then remove the ones from `server` and to use only these ones?

Reply:
Good idea, pushed a commit doing this!